enhancements/E-0001-secret-handling.md

```diff
@@ -80,6 +80,7 @@ What is out of scope for this enhancement proposal?
 Listing non-goals helps to focus discussion and make progress.
 -->
 - Providing facilities to automagically inject (pull) secret references into managed objects. (For example PodSpecs or (Cluster)Packages.)
+- Syncing secrets across the MC/HC barried in the HyperShift architecture.
```

Comment:
Since this was brought up yesterday, and I wanted to cross-post here for posterity: in theory this might work without any extra effort out of the box since _controlplane_ and _dataplane_ can talk to each other via _konnectivity_ proxy?

Reply:
This would even work without the konnectivity service because  workloads in the MC can directly talk to the HC control-planes hosted in that cluster.

I ruled that functionality out bc of:
- architecture concerns: PKO in the MC does not talk to HC control-planes directly as of today
- scalability concerns: PKO needs to be able to concurrenctly  watch and reconcile the synced secret objects in many hosted control-planes at the same time
- security concerns: Not providing generic capabilities to sync secrets from MC to HC also does not provide the ability to accidentally expose the wrong (platform) secrets to a hosted-cluster
- roadmap concerns: we weren't asked for the capabilities.

Do you think we need this feature?